### ISSUE TITLE
Maintain camera heading on zoom

### ISSUE THREAD

**Issue by rutsky** [AnalyticalGraphicsInc/cesium#4639]
By default panning Earth in 3D has nice property that no matter how I pan tiles on Earth surface will always be oriented "vertically":

![earth-panning-optimized](https://cloud.githubusercontent.com/assets/46573/20227963/88f3800a-a868-11e6-9560-1aff07e8d37b.gif)

When zoom to mouse is used (either using mouse wheel, or by holding right mouse button), Earth axis is being tilted (the further away from screen center zooming is done, the stronger axis tilt):

![earth-zoom-to-mouse2](https://cloud.githubusercontent.com/assets/46573/20228353/4a44bdd6-a86a-11e6-8497-0f25c4df6b42.gif)

If I disable zooming to mouse position (by hardcoding in Cesium source code) or if I zoom exactly to screen center this issue is not reproduced.

This is very inconvenient when you navigate over Earth using panning and zooming and at some point see that tiles on the map are rotated.

Also mentioned here: https://groups.google.com/d/msg/cesium-dev/Kv_dSET4Bjs/Z0XJp1JIBQAJ

**Comment by hpinkos** [comment]
Thanks @rutsky,  this is something that has come up a few times on the forum as well. 
Until we address this, you can add a `preRender` event to always keep the camera heading such that north is up.  Here is an example: 

``` javascript
var viewer = new Cesium.Viewer('cesiumContainer');
var camera = viewer.scene.camera;
viewer.scene.preRender.addEventListener(function() {
    camera.setView({
        orientation: {
            heading: 0,
            pitch: camera.pitch,
            roll: camera.roll
        }
    });
});
```

**Comment by rutsky** [comment]
@hpinkos thanks for the workaround!

**Comment by pjcozzi** [comment]
CC #2816

**Comment by wallw-bits** [comment]
I am attempting a fix for this, but I'm a little stuck.

My attempted solution was to grab the heading, pitch, and roll from the camera at  `Source/Scene/ScreenSpaceCameraController.js:488` and then apply them back via `camera.setView(...)` at the end of the function. This works great and keeps the heading at 0 (or 360), except that if you zoom in far enough, something outside of that function modifies the camera heading to a non-zero value (haven't checked pitch and roll yet). I can consistently reproduce this using the viewer and only the mouse wheel to zoom. Still trying to track down what is causing that.

**Comment by wallw-bits** [comment]
My employer is working on a corporate CLA. Should be good to go on Monday and I'll open the PR.

**Comment by pjcozzi** [comment]
@wallw-bits thanks for working on the fix, looking forward to it!

**Comment by cesium-concierge** [comment]
Congratulations on closing the issue! I found these Cesium forum links in the comments above:

https://groups.google.com/d/msg/cesium-dev/Kv_dSET4Bjs/Z0XJp1JIBQAJ

If this issue affects any of these threads, please post a comment like the following:

> The issue at https://github.com/AnalyticalGraphicsInc/cesium/issues/4639 has just been closed and may resolve your issue. Look for the change in the next stable release of Cesium or get it now in the master branch on GitHub https://github.com/AnalyticalGraphicsInc/cesium.


__I am a bot who helps you make Cesium awesome!__ Thanks again.

**Comment by pjcozzi** [comment]
Thanks again for the bug report, @rutsky!  This was fixed in #5603.
